include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -391,9 +376,9 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
                     header_entry = std::addressof(hdr.grouping);
                     break;
                 }
-                default: // unsupported header tag
+                default: // unknown/user tag
                 {
-                    consume_unsupported_tag_and_print_warning("HD", raw_tag);
+                    parse_and_append_unhandled_tag_to_string(hdr.user_tags, raw_tag);
```

Comment:
Alla the header stuff (SQ, HD, RG) have strutcs that store the members defined in the SAM spec. Any user defined tags are stored as string. For the alignment records, tags, including User-defined, are stored in the SAM tag dictionary.

Reply:
Yes, but do I see it right that here we store the user defined field tags in a tab seperated string? If so, I feel like it would make more sense to put this into a vector of field tags instead.

Reply:
Sounds good! Would you mind tracking this on the issue board?

Reply:
Will do when working on this issue

Reply:
https://github.com/seqan/seqan3/issues/3283